app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt

```diff
@@ -16,722 +16,723 @@
+       .catch { it.printStackTrace() }
+       .onEach { books -> bookDao.insert(books) }
+       .flowOn(dispatcher)
+       .launchIn(viewModelScope)
+ 
+   private fun fileSelectActions() =
+     fileSelectActions
+       .onEach { action ->
+         runCatching {
+           sideEffects.emit(
+             when (action) {
+               is RequestNavigateTo -> OpenFileWithNavigation(action.bookOnDisk)
+               is RequestMultiSelection -> startMultiSelectionAndSelectBook(action.bookOnDisk)
+               RequestDeleteMultiSelection -> DeleteFiles(selectionsFromState(), alertDialogShower)
+               RequestShareMultiSelection -> ShareFiles(selectionsFromState())
+               MultiModeFinished -> noSideEffectAndClearSelectionState()
+               is RequestSelect -> noSideEffectSelectBook(action.bookOnDisk)
+               RestartActionMode -> StartMultiSelection(fileSelectActions)
+               UserClickedDownloadBooksButton -> NavigateToDownloads
+             }
+           )
+         }.onFailure {
+           it.printStackTrace()
+         }
+       }.launchIn(viewModelScope)
+ 
+   private fun startMultiSelectionAndSelectBook(
+     bookOnDisk: BookOnDisk
+   ): StartMultiSelection {
+     fileSelectListStates.value?.let {
+       fileSelectListStates.postValue(
+         it.copy(
+           bookOnDiskListItems = selectBook(it, bookOnDisk),
+           selectionMode = MULTI
+         )
+       )
+     }
+     return StartMultiSelection(fileSelectActions)
+   }
+ 
+   private fun selectBook(
+     it: FileSelectListState,
+     bookOnDisk: BookOnDisk
+   ): List<BooksOnDiskListItem> {
+     return it.bookOnDiskListItems.map { listItem ->
+       if (listItem.id == bookOnDisk.id) {
+         listItem.apply { isSelected = !isSelected }
+       } else {
+         listItem
+       }
+     }
+   }
+ 
+   private fun noSideEffectSelectBook(bookOnDisk: BookOnDisk): SideEffect<Unit> {
+     fileSelectListStates.value?.let {
+       fileSelectListStates.postValue(
+         it.copy(bookOnDiskListItems = selectBook(it, bookOnDisk))
+       )
+     }
+     return None
+   }
+ 
+   private fun selectionsFromState() = fileSelectListStates.value?.selectedBooks.orEmpty()
+ 
+   private fun noSideEffectAndClearSelectionState(): SideEffect<Unit> {
+     fileSelectListStates.value?.let {
+       fileSelectListStates.postValue(
+         it.copy(
+           bookOnDiskListItems =
+             it.bookOnDiskListItems.map { booksOnDiskListItem ->
+               booksOnDiskListItem.apply { isSelected = false }
+             },
+           selectionMode = NORMAL
+         )
+       )
+     }
+     return None
+   }
+ 
+   @OptIn(ExperimentalCoroutinesApi::class)
+   private fun requestsAndConnectivityChangesToLibraryRequests(
+     library: MutableSharedFlow<LibraryNetworkEntity>,
+     dispatcher: CoroutineDispatcher = Dispatchers.IO
+   ) = requestDownloadLibrary.flatMapConcat {
+     connectivityBroadcastReceiver.networkStates.asFlow()
```

Comment:
`connectivityBroadcastReceiver.networkStates` is already a `flow`, why are you again converting this to `flow`.

Now you have converted the `networkStates` into `StateFlow` so applying 'distinctUntilChanged' to StateFlow has no effect.  Please remove it.